### DISCUSSION TITLE
[11 gennaio 2022 - A] Esercizio 2

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#194] [Esami]
**Esercizio 2 (2+2+1+2 punti)**

- Si consideri il circuito in figura e si scriva l’espressione della funzione _f_
- Trasformare tale espressione, usando assiomi e regole dell’algebra di Boole, in forma normale SOP
- Si stenda la tavola di verità di _f_
- Si scrivano le espressioni minimali SOP e POS di _f_

![image](https://user-images.githubusercontent.com/69389461/211819404-6ed60171-0a8b-45da-92c6-5695502dfbbb.png)

**Comment by Elia-Belli** [comment]
- $f (A.B.C.D) = (B \bar C + \bar B A) \oplus (D(AC + \bar AD))$
- $(\overline {B \bar C})(\overline {\bar B A})(ACD + \bar A D) + (B \bar C + \bar B A)(\overline {ACD})(\overline {\bar A D})$
$(\bar B + C)(B+\bar A)(ACD + \bar A D)+(B \bar C + \bar B A)(\bar A + \bar C + \bar D)(A + \bar D)$
$(\bar A \bar B + BC + \bar A C)(ACD + \bar A D) +(B \bar C + AB)(\bar A \bar D + A \bar C + \bar C \bar D + A \bar D + \bar D)$
$(\bar A \bar B D + ABCD + \bar A CD) + (AB \bar C +B \bar C \bar D + A\bar B \bar C+A \bar B \bar D)$
$\bar A \bar B D + ABCD + \bar A CD + AB \bar C +B \bar C \bar D + A\bar B \bar C+A \bar B \bar D$


- Tabella della verità e K-map per ottenere le espressioni minimali
$Y = A\bar C + B \bar C \bar D + A \bar B \bar D + \bar A \bar B D + BCD$
$Y = (A+B+D)(A+ \bar B + C + \bar D)(\bar A + B + \bar C + \bar D)(\bar B + \bar C + D)$
![Immagine 2023-01-17 184547](https://user-images.githubusercontent.com/107944476/212974048-31b22e33-fa63-444c-8746-9893ae706782.png)

**Comment by Baliano1999** [comment]
Credo ci sia un errore: nella seconda riga, dopo aver negato (AB' + C'B)  non è stato negato il termine "A"

diventerebbe: (B' + C) (B + A') (ACD + DA') + (AB' + C'B) ( ( C' + A') (D' + A) + D') 


![unnamed](https://user-images.githubusercontent.com/122894801/212963549-673415e3-0939-4b4b-ab41-4f57e2c442fa.jpg)

Fatemi sapere se vi trovate e se sto sbagliando io :)

**Comment by Elia-Belli** [comment]
@Baliano1999 grazie, ho corretto un paio di errori e adesso torna
Anche nella tua c'è almeno un errore alla riga 3-terza parentesi, facendo il prodotto scrivi $\bar A B$ anzichè $\bar A \bar B$ alla riga successiva

**Comment by Baliano1999** [comment]
Azz, hai ragione ho sbagliato. 

Quel prodotto fortunatamente non cambia il risultato finale, poichè successivamente non viene moltiplicato con nessun altro elemento.

Per me ora è tutto corretto.

**Comment by Elia-Belli** [comment]
@Baliano1999 $(\bar A C +\bar A \bar B + BC)ACD +D \bar A$ è diverso da $(\bar A C +\bar A \bar B + BC)(ACD +D \bar A)$

**Comment by Baliano1999** [comment]
Perdonami non sto capendo dove sia l'errore, potresti mostrarmelo?

**Comment by Baliano1999** [comment]
Non mi è chiaro come risolvi 

![domanda](https://user-images.githubusercontent.com/122894801/212978014-ac40bd05-1a2e-40b6-9e5a-2f0914b8ced5.PNG)

non capisco perchè poi diventi cosi, non manca un A'BCD ?

![domanda](https://user-images.githubusercontent.com/122894801/212978281-035b23c1-6b73-4ed7-94e2-286487063214.PNG)

**Comment by Elia-Belli** [comment]
$(\bar A C +\bar A \bar B + BC)(ACD +D \bar A)$
$(\bar A ACD + \bar A CD + \bar A A\bar B CD + \bar A \bar B D + ABCD + \bar A BCD)$
$(0+ \bar A CD +0+ \bar A \bar B D + ABCD + \bar A BCD)$
se assorbo $\bar A BCD$ con $\bar A CD$ ottengo
$(\bar A CD + \bar A \bar B D + ABCD)$

**Comment by Baliano1999** [comment]
Pensavo che l'assorbimento in questo modo non fosse possibile

Posso chiederti come inserisci lo xor in https://www.boolean-algebra.com/ ?

**Comment by Elia-Belli** [comment]
Beh sì è possibile basta vedere che se raccogli esce questo $\bar A CD (1+B) = \bar A CD (1) = \bar A CD$
Per lo xor non so nemmeno se sia possibile, di solito lo riscrivo con solo OR e AND

**Comment by Baliano1999** [comment]
Scusami ma non converrebbe mettere in evidenza BCD, ovvero:

A'CD+A'B'D+ABCD+A'BCD
A'CD+A'B'D+BCD(A+A')
A'CD+A'B'D+BCD

**Comment by Baliano1999** [comment]
@Elia-Belli fammi sapere se condividi o meno

**Comment by Elia-Belli** [comment]
Potresti farlo, io ho fatto la cosa più istintiva che mi veniva in mente per arrivare alla forma SOP (in realtà avrei potuto lasciare sia $ABCD$ che $\bar A BCD$), alla fine è solo una semplificazione in più, in ogni caso quando minimizzi con le k-map tornerai allo stesso risultato qualunque cosa tu faccia (salvo prendere implicanti primi diversi nella k-map)

**Comment by fraxb04** [comment]
@Elia-Belli 
posso chiederti se hai usato shannon per trovare l'equazione del circuito? 
ps: il punto 3, 4 sono giusti

**Comment by Elia-Belli** [comment]
@fraxb04 nope, ho solo trascritto le porte logiche con simboli

**Comment by fraxb04** [comment]
@Elia-Belli 
ho controllato meglio la SOP minimizzata e !B!C!D in realtà è uguale a B!C!D
edit : è tutto corretto tranne ⬆

**Comment by FeddyLix17** [comment]
@Elia-Belli prima di sviluppare lo xor, **(D(AC+ A&#x0304;D))** poteva essere semplificato in questo modo?
(D(AC + A&#x0304;D))
(ACD + A&#x0304;DD)
(ACD + A&#x0304;D)
(D(AC + A&#x0304;))
(D(C + A&#x0304;))
(CD + A&#x0304;D)
